### PR TITLE
Cron interval

### PR DESCRIPTION
**Pull request by Nef10**
Instead of the next enqueuing the concrete interval is now shown:

pasted image at 2016_05_30 11_34 am

This depends on travis-ci/travis-api#266.

### CODE REVIEW

File: app/templates/components/cron-job.hbs [hunk 1]
```diff
@@ -1,6 +1,10 @@
 <div class="cron-job-value branch-name">{{cron.branch.name}}</div>
 <div class="cron-job-value interval">{{cron.interval}}</div>
-<div class="cron-job-value next-enqueuing">{{pretty-date cron.next_enqueuing}}</div>
+{{#if isOverdue}}
+  <div class="cron-job-text next-enqueuing">{{"soon"}}</div>
```
Comment (clekstro) [inline app/templates/components/cron-job.hbs]: If we're just returning a string here, no need for the `{{ }}` syntax. We can just do:

```
<div class="cron-job-text next-enqueuing">soon</div>
```

File: app/components/cron-job.js [hunk 2]
```diff
@@ -9,8 +9,8 @@ export default Ember.Component.extend({
   store: service(),
 
   isOverdue: function(key) {
```
Comment (clekstro) [inline app/components/cron-job.js]: These computed properties don't appear to be using the `key` being passed in. Can they be removed?

File: app/components/cron-job.js [hunk 3]
```diff
@@ -8,12 +8,39 @@ export default Ember.Component.extend({
...
+        weekday[5] = "Friday";
+        weekday[6] = "Saturday";
+        text = 'Every ' + weekday[creationTime.getDay()] + ' at ' + creationTime.getHours();
+        break;
+      case 'daily':
+        text = 'Every day at ' + creationTime.getHours();
+        break;
     }
-    return false;
+    return text;
```
Comment (clekstro) [inline app/components/cron-job.js]: It might be nice to break this up into some smaller functions to make it a little more readable.

File: tests/acceptance/repo-settings-test.js [hunk 4]
```diff
@@ -111,13 +111,11 @@ test('view settings', function(assert) {
...
     assert.equal(settingsPage.crons(0).branchName, 'daily-branch');
-    assert.equal(settingsPage.crons(0).interval, 'daily');
-    assert.ok(settingsPage.crons(0).nextEnqueuing.indexOf(':19'), 'expected cron next enqueuing to match');
+    assert.ok(settingsPage.crons(0).nextEnqueuing.indexOf('Every') === 0, 'expected cron next enqueuing to match');
     assert.ok(settingsPage.crons(0).disableByBuildText.indexOf('Even') === 0, 'expected cron to run even if no new commit after last build');
 
     assert.equal(settingsPage.crons(1).branchName, 'weekly-branch');
-    assert.equal(settingsPage.crons(1).interval, 'weekly');
-    assert.ok(settingsPage.crons(1).nextEnqueuing.indexOf(':11'), 'expected cron next enqueuing to match');
+    assert.ok(settingsPage.crons(1).nextEnqueuing.indexOf('Every') === 0, 'expected cron next enqueuing to match');
```
Comment (clekstro) [inline tests/acceptance/repo-settings-test.js]: Can we distinguish between the assertion messages in these two cases?  Maybe something like "Shows weekly/daily enqueuing text" or something to that effect?

**Comment by clekstro** [comment]
Except for the few things that could use a little cleanup, code looks fine to me 👍 .

**Comment by Nef10** [comment]
@clekstro adjusted according to your comments

File: app/components/cron-job.js [hunk 5]
```diff
@@ -8,6 +8,46 @@ export default Ember.Component.extend({
   actionType: 'Save',
   store: service(),
 
+  intervalText: function() {
```
Comment (renee-travisci) [inline app/components/cron-job.js]: I would split this up into 3 helper methods for extracting the time, calculating the day of the month, and getting the day of the week. Then have the procedural function that is putting together the text variable, call your helper functions. The encapsulation will help test and isolate logic, which is easier to maintain for 3 reasons: 

1. Smaller procedural functions are easier to read and understand for other developers (or yourselves) in the future. 

2. You can unit test the isolated logic (it's easier to cover edge cases of smaller logic).

3. You can replace the logic more easily if needed in the future. For example if we wanted to support another language we could edit the function that grabs weekday names, and add new tests for just that logic without changing the larger procedural function or its integration tests.

**Comment by renee-travisci** [comment]
As an FYI there is a large js library that does this type of work that you may want to look at for reference called later.js (https://bunkat.github.io/later/). I don't think we would want to bring in such a large dependency for something simple like this, but it might be a good learning tool to look at that code.

File: app/components/cron-job.js (hunk 5 above)
Comment (clekstro) [inline app/components/cron-job.js]: @renee-travisci Thanks for such clear instructions. I had mentioned breaking this up before, but this is much more actionable.

**Comment by chistopher** [comment]
changed text 
image

**Comment by chistopher** [comment]
reopen
